apps/gallery/js/MetadataParser.js

```diff
@@ -303,7 +303,16 @@ var metadataParser = (function() {
         // thumbnail and an external preview image. Note that progressive
         // jpegs will have a type 'pjpeg' and will be handled below
         // like png images.
-        if (metadata.type === 'jpeg') {
+
+        // For jpeg images with sizes more than 2.5 mb, we need
+        // to create preview on sdcard to avoid flash while opening image.
+        // See bug 1181290.
+        // This constraint can be removed when gecko supports downscale during
```

Comment:
We don't know for sure that we'll be able to remove the constraint. We hope so, though.